pkgs/applications/editors/spacevim/default.nix

```diff
@@ -0,0 +1,60 @@
+{ makeWrapper, vim_configurable, vimPlugins, fetchFromGitHub, writeTextDir
+, stdenv, spacevim_toml ? "" }:
+with stdenv;
+let
+  # At the moment, there is no way to write TOML given a attrset
+  # so the solution here is to allow a custom TOML file to be provided.
+  # The pkg includes a sensible TOML file for those that don't specify one.
+  config = if builtins.stringLength spacevim_toml == 0 then
+    "${builtins.readFile ./init.toml}"
+  else
+    spacevim_toml;
+  vim-customized = vim_configurable.customize {
+    name = "vim";
+    # Not clear at the moment how to import plugins such that
+    # SpaceVim finds them and does not auto download them to
+    # ~/.cache/vimfiles/repos
+    vimrcConfig.packages.myVimPackage = with vimPlugins; { start = [ ]; };
+  };
+  spacevimdir = writeTextDir "/init.toml" config;
+
+  spacevim = mkDerivation {
+    name = "spacevim";
+    src = fetchFromGitHub {
+      owner = "SpaceVim";
+      repo = "SpaceVim";
+      rev = "c937c0e2fd37207c36c8c5e53b36c41d7222fee6";
```

Comment:
You should be using the latest release of spacevim (1.4.0 of writing this) and not a random commit. You would also need to add the `version` field, switch from `name` to `pname` and update the PR to match the version.

Reply:
@skykanin their install script installs it off of master.
I don't think they are making releases that often.

What's guidance here ?
Looks like the commit SHA is more appropriate.

I tried setting **pname** but it didn't work (maybe only works when version present?)

Reply:
I'm not completely sure what the standard practice is for versioning, but personally I would use the latest official release and if you want the bleeding edge version from the master branch override the package build locally. Indeed `pname` does require a `version` because it combines them into `"${pname}-${version}"`, [from the manual](https://nixos.org/nixpkgs/manual/#sec-using-stdenv).

Reply:
`name` should indeed not be used anymore. When using a commit (rather than a release tag), there are still diverging conventions, but they all seem to share that the version should contain the commit date in the form `YYYY-MM-DD` and that `unstable` should be used somewhere. My preferred flavor:

```
pname = "spacevim";
version = "unstable-YYYY-MM-DD"
```

(where you fill `YYYY-MM-DD`).

I also agree with @skykanin  that, unless the latest release is very old, using a tagged version is highly preferred.

Reply:
Thanks -- I've changed to match your suggestions.
The tagged version _is really old_; so i've opted to change to _unstable_ methodology you've prescribed.

Reply:
@fzakaria @danieldk the `-unstable` suffix belongs to `pname`. See #68518.

Reply:
Looks like it hasn't been decided yet @Ma27 ; hoping we can just come back to this afterwards. 
It will be a small fixup.

Reply:
my argument was that changing the pname essentially "changes the package" as far as repology is concerned, which is weird to have packages move in and out of unique categorization. Having the version contain `unstable` at least keeps the package name the same.

Then again, this convention was first used when `name = "pkg-${version}";` was still very common, and there wasn't a clear delineation between pname and version established.

Reply:
As an outsider: I'm in favor as a user of having `pname` be the package name; _unstable_ feels itself like a version moniker.